src/main/scala/mesosphere/marathon/api/v2/validation/PodsValidation.scala

```diff
@@ -64,9 +65,84 @@ trait PodsValidation {
+    hc.endpoint should matchRegexFully(NamePattern)
+    hc.endpoint is isTrue("contained in the container endpoints") { endpoint =>
+      endpoints.exists(_.name == endpoint)
+    }
+    hc.path.map(_.length).getOrElse(1) is between(1, 1024)
+  }
+
+  def tcpHealthCheckValidator(endpoints: Seq[Endpoint]) = validator[TcpHealthCheck] { hc =>
+    hc.endpoint.length is between(1, 63)
+    hc.endpoint should matchRegexFully(NamePattern)
+    hc.endpoint is isTrue("contained in the container endpoints") { endpoint =>
+      endpoints.exists(_.name == endpoint)
+    }
+  }
+
+  val commandCheckValidator = new Validator[CommandHealthCheck] {
+    override def apply(v1: CommandHealthCheck): Result = v1.command match {
+      case ShellCommand(shell) =>
+        (shell.length should be > 0)(shell.length)
+      case ArgvCommand(argv) =>
+        (argv.size should be >0)(argv.size)
+    }
+  }
+
+  def healthCheckValidator(endpoints: Seq[Endpoint]) = validator[HealthCheck] { hc =>
+    hc.gracePeriodSeconds should be >= 0L
+    hc.intervalSeconds should be >= 0
+    hc.maxConsecutiveFailures should be >= 0
+    hc.timeoutSeconds should be >= 0
+    hc.delaySeconds should be >= 0
+    hc.http is optional(httpHealthCheckValidator(endpoints))
+    hc.tcp is optional(tcpHealthCheckValidator(endpoints))
+    hc.command is optional(commandCheckValidator)
+    hc is isTrue("Only one of http, tcp, or command may be specified") { hc =>
+      hc.http.isDefined ^ hc.tcp.isDefined ^ hc.command.isDefined
+    }
+  }
+
+  val endpointValidator = validator[Endpoint] { endpoint =>
+    endpoint.name.length is between(1, 63)
+    endpoint.name should matchRegexFully(NamePattern)
+    endpoint.containerPort.getOrElse(1) is between(1, 65535)
+    endpoint.hostPort.getOrElse(0) is between(0, 65535)
+    endpoint.protocol.size should be > 0
+  }
+
+  val imageValidator = validator[Image] { image =>
+    image.id.length is between(1, 1024)
+  }
+
+  // TODO(PODS): don't we need to know what path we're mounting, not just where it is in the container?
+  // kind of confused by this one.
+  @SuppressWarnings(Array("UnusedParameter"))
+  def volumeMountValidator(volumes: Seq[Volume]): Validator[VolumeMount] = validator[VolumeMount] { volumeMount => // linter:ignore:UnusedParameter
+    volumeMount.name.length is between(1, 63)
+    volumeMount.name should matchRegexFully(NamePattern)
+    volumeMount.mountPath.length is between(1, 1024)
+  }
+
+  val artifactValidator = validator[Artifact] { artifact =>
+    artifact.uri.length is between(1, 1024)
+    artifact.destPath.map(_.length).getOrElse(1) is between(1, 1024)
+  }
+
+  val lifeCycleValidator = validator[Lifecycle] { lc =>
+    lc.killGracePeriodSeconds.getOrElse(0.0) should be > 0.0
+  }
+
+  def containerValidator(volumes: Seq[Volume]): Validator[PodContainer] = validator[PodContainer] { container =>
     container.resources is valid(resourceValidator)
+    container.endpoints is empty or every(endpointValidator)
```

Comment:
`every` checks every object in the seq. Is there additional value in `is empty or` ?